sematic/resolvers/local_resolver.py

```diff
@@ -446,9 +505,27 @@ def _populate_graph(
         Update the graph based on future.
         """
         if future.id not in self._runs:
+            # TODO: the run is being created as a side effect of updating the graph;
+            # the run and future should be atomically updated, if not the same concept;
+            # the run lifecycle should be the main flow the resolver deals with, and the
+            # graph should be a consequence of this, without being updated through lateral
+            # effect such as the run is now, and without the disjointed partial save from
+            # below
```

Comment:
There is no single place where `Future`s or `Run`s are created once the `Resolution` has started. Their lifecycles are also disjointed.